src/NHSD.BuyingCatalogue.Organisations.Api/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -14,7 +14,7 @@ public static IServiceCollection RegisterHealthChecks(this IServiceCollection se
             if (connectionString is null)
                 throw new ArgumentNullException(nameof(connectionString));
 
-            if (smtpSettings is null) 
+            if (smtpSettings is null)
```

Comment:
Would adding the NotNull attribute to the parameter work here? I've not messed with it much, but I think it's purpose is to get rid of this kind of thing

Reply:
The `[NotNull]` attribute was introduced in 3.0 as part of the `System.Diagnostics.CodeAnalysis` namespace but I'm not yet sure of its correct/intended usage. It doesn't appear to be that well documented but it appears to be more about outputs than inputs. For example, from the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullattribute?view=netcore-3.1):

> Specifies that an input argument was not null when the call returns.

I don't believe it will do anything to prevent a null parameter being passed in, although it's probably worth investigating the correct/intended usage.

The only other `[NotNull]` attribute I'm aware of is the R# annotation but that obviously only works with ReSharper so it may not be something we want to rely on.